legal-api/src/legal_api/resources/business/business_filings.py

```diff
@@ -501,9 +501,12 @@ def _create_invoice(business: Business,  # pylint: disable=too-many-locals
         if filing.filing_type == Filing.FILINGS['incorporationApplication'].get('name'):
             mailing_address = Address.create_address(
                 filing.json['filing']['incorporationApplication']['offices']['registeredOffice']['mailingAddress'])
-            corp_type = filing.json['filing']['incorporationApplication']['nameRequest']['legalType']
-            business.legal_name = filing.json['filing']['incorporationApplication']['nameRequest'].get('legalName') \
-                or business.identifier
+            corp_type = filing.json['filing']['business'].get('legalType', 'BC')
+
+            try:
+                business.legal_name = filing.json['filing']['incorporationApplication']['nameRequest']['legalName']
+            except KeyError:
+                business.legal_name =  business.identifier
```

Comment:
What's the issue you're trying to fix here? I don't think this will change anything, but it looks like there could be a problem where the 'business.identifier' doesn't exist yet for an FED incorp? When a future effective numbered company is filed there's no legal name and no business yet

Reply:
Okay I just saw your schema PR -> will the business.identifier exist already or does it need to be filing.json['filing']['business']['identifier']?

Reply:
I believe the business Identifier in this case will come from the bootstrap entity (it is a union type way up in the initial caller...from what I recall)